### PR TITLE
 add multi-component version fallback

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java [hunk 1]
```diff
@@ -190,8 +186,7 @@ public APIResponseDTO getPartialComTrans(String productName,
                HttpServletRequest req) throws Exception {
           
           TranslationDTO resulttranslationDTO = getResultTranslationDTO( productName, version,components, locales,  pseudo, req);
-          TranslationDTO allTranslationDTO  =  getAllCompTrans( productName,  version, pseudo, req);
-        
+          TranslationDTO allTranslationDTO  =  getAllCompTrans( resulttranslationDTO.getProductName(),  resulttranslationDTO.getVersion(), pseudo, req);
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java]: Should pseudo also from resulttranslationDTO? It's better to aligh with parameters 'product' and 'version'.

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java [hunk 2]
```diff
@@ -33,4 +37,97 @@ public static boolean checkParams(String[] categories, String... args){
...
+                    }
+                }
+            }
+        }
+        return StringUtils.isEmpty(mv)? version : mv;
+    }
+    
+    
+    /**
+     * Compare source version and target version
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: Is this copied? I think it's not a good pratice to have same code in different places.

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java [hunk 3]
```diff
@@ -103,13 +104,7 @@ private TranslationDTO getAllCompTrans(String productName, String version,String
     
     public APIResponseDTO getMultTrans(String productName, String version, String componentsStr, String localesStr, String pseudo,
             HttpServletRequest req)  throws Exception {
-        if(StringUtils.isEmpty(componentsStr) && StringUtils.isEmpty(localesStr)) {
-            return  super.handleResponse(APIResponseStatus.OK, getAllCompTrans( productName,  version, pseudo, req)) ;
-        }else {
             return getPartialComTrans(productName, componentsStr,  version,  localesStr,  pseudo, req);
-        }
-        
-
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java]: Why did you remove these lines? What if StringUtils.isEmpty(componentsStr) && StringUtils.isEmpty(localesStr) ?

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java (hunk 1 above)
Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java]: done

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java (hunk 3 above)
Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java]: As Colin's recommend change the design. the new design have this feature

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java (hunk 2 above)
Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: Because Colin's method is private, I can't use it. If I want to use it, I need to change his codes

Comment (Xiaochao8) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: Could we refactor the code to make public or move to common class? So we can avoid duplication.

Comment (tigershi) [reply g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: done

File: g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java [hunk 4]
```diff
@@ -6,9 +6,13 @@
...
 import com.vmware.vip.common.constants.ConstantsKeys;
+import com.vmware.vip.core.messages.exception.L3APIException;
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
 
 public class CommonUtility {
```
Comment (linr211) [inline g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/utils/CommonUtility.java]: CommonUtility is too general, change it to VersionMatcher?
